### ISSUE TITLE
Compiling autogenerated code with -Wextra

### ISSUE THREAD

**Issue by laudrup** [cinemast/libjson-rpc-cpp#71]
I prefer to compile me code with the -Wextra option (among others) warnings and it is almost possible with the autogenerated stubs from the jsonrpcstub program.

The only issue is, that if an autogenerated method doesn't accept any parameters, GCC (and possibly others) will warn that "request" is an unused parameter.

I find the "unused parameter" warning fairly useful, so I would prefer to keep that warning.

I have worked around it by adding `this->cg.writeLine("(void)request;");` at the start of the function body generation in cppserverstubgenerator.cpp, but I'm not sure this is the nicest solution.

I could also hack around my build scripts to build the stubgenerator with less warnings enabled, but that would be a bit of a shame.

So in short, it seems like the code can _almost_ compile with -Wextra and I personally find it worth the minor effort to make it able to compile cleanly with -Wextra.

Thanks.
